{"uid":"44489cab4c8d9510","name":"test_record_d365_authentication","fullName":"tests.d365.test_d365_auth#test_record_d365_authentication","historyId":"92b47c5c72be6742c31b5c1edf35d301","time":{"start":1760633823790,"stop":1760633944536,"duration":120746},"description":"\n    Record D365 authentication session.\n    \n    IMPORTANT: Run this test manually with headed mode:\n        pytest tests/d365/test_d365_auth.py -k record --headed -s\n    \n    Steps:\n        1. Navigate to D365\n        2. Login manually\n        3. Wait for dashboard to load\n        4. Session is saved automatically\n    ","descriptionHtml":"<pre><code>Record D365 authentication session.\n\nIMPORTANT: Run this test manually with headed mode:\n    pytest tests/d365/test_d365_auth.py -k record --headed -s\n\nSteps:\n    1. Navigate to D365\n    2. Login manually\n    3. Wait for dashboard to load\n    4. Session is saved automatically\n</code></pre>\n","status":"failed","statusMessage":"Failed: Authentication timeout. Please login within 2 minutes. Error: Not on D365 dashboard yet. Current URL: https://login.microsoftonline.com/ba4f0c17-6451-450e-8adb-a7df399faaa9/oauth2/authorize?client_id=00000015-0000-0000-c000-000000000000&response_type=code%20id_token&scope=openid%20profile&state=OpenIdConnect.AuthenticationProperties%3DAQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAKa_EB3DXdU6q3ruxEZ2yZAAAAAACAAAAAAAQZgAAAAEAACAAAABOWze7vCAhxSiq7P-B-qGuQI75QkTlc94ZpQVyFl5J7gAAAAAOgAAAAAIAACAAAABGu2Yvlo1Ifgl9rF4IkIVSrc31luJ2iQeXfG-RtfnzxHAAAACO4ZTdzoP0qHmQa2agUPhI7ZuxaSFx4gcv77g4RoUyDrTLZEW2UoVc8Qjx0i1ND2JdrMqiGASgI8lrCXWL97uR4Atvsn74ku62IQ4OlDtQDQiY9Pu5P7TQ108bXPd47EvmyKMFUjapBbUgulRkTBpoQAAAAAMeTlLhmw6StoYoUVE_cv3F2num1C1MS7focWGvftNSNeRnoaE8vwsDERDou5Ty77VmSXKOdUKkhsM1WemM5CM%26RedirectTo%3Dhttps%253A%252F%252Ffourhands-test.sandbox.operations.dynamics.com%252F&response_mode=form_post&nonce=638962306249100599.Zjc2MDkwNDEtYTQxMC00MjY4LThkN2YtMTI0MWM4ODUzZWQwYzVkZmZlMDQtMmM0OC00YjI1LTlkYzktM2M4OWY4OTEwZWYx&redirect_uri=https%3A%2F%2Ffourhands-test.sandbox.operations.dynamics.com%2F&max_age=86400&x-client-SKU=ID_NET472&x-client-ver=8.3.0.0","statusTrace":"tests/d365/test_d365_auth.py:55: in test_record_d365_authentication\n    raise Exception(f\"Not on D365 dashboard yet. Current URL: {current_url}\")\nE   Exception: Not on D365 dashboard yet. Current URL: https://login.microsoftonline.com/ba4f0c17-6451-450e-8adb-a7df399faaa9/oauth2/authorize?client_id=00000015-0000-0000-c000-000000000000&response_type=code%20id_token&scope=openid%20profile&state=OpenIdConnect.AuthenticationProperties%3DAQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAKa_EB3DXdU6q3ruxEZ2yZAAAAAACAAAAAAAQZgAAAAEAACAAAABOWze7vCAhxSiq7P-B-qGuQI75QkTlc94ZpQVyFl5J7gAAAAAOgAAAAAIAACAAAABGu2Yvlo1Ifgl9rF4IkIVSrc31luJ2iQeXfG-RtfnzxHAAAACO4ZTdzoP0qHmQa2agUPhI7ZuxaSFx4gcv77g4RoUyDrTLZEW2UoVc8Qjx0i1ND2JdrMqiGASgI8lrCXWL97uR4Atvsn74ku62IQ4OlDtQDQiY9Pu5P7TQ108bXPd47EvmyKMFUjapBbUgulRkTBpoQAAAAAMeTlLhmw6StoYoUVE_cv3F2num1C1MS7focWGvftNSNeRnoaE8vwsDERDou5Ty77VmSXKOdUKkhsM1WemM5CM%26RedirectTo%3Dhttps%253A%252F%252Ffourhands-test.sandbox.operations.dynamics.com%252F&response_mode=form_post&nonce=638962306249100599.Zjc2MDkwNDEtYTQxMC00MjY4LThkN2YtMTI0MWM4ODUzZWQwYzVkZmZlMDQtMmM0OC00YjI1LTlkYzktM2M4OWY4OTEwZWYx&redirect_uri=https%3A%2F%2Ffourhands-test.sandbox.operations.dynamics.com%2F&max_age=86400&x-client-SKU=ID_NET472&x-client-ver=8.3.0.0\n\nDuring handling of the above exception, another exception occurred:\ntests/d365/test_d365_auth.py:58: in test_record_d365_authentication\n    pytest.fail(f\"Authentication timeout. Please login within 2 minutes. Error: {e}\")\nE   Failed: Authentication timeout. Please login within 2 minutes. Error: Not on D365 dashboard yet. Current URL: https://login.microsoftonline.com/ba4f0c17-6451-450e-8adb-a7df399faaa9/oauth2/authorize?client_id=00000015-0000-0000-c000-000000000000&response_type=code%20id_token&scope=openid%20profile&state=OpenIdConnect.AuthenticationProperties%3DAQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAKa_EB3DXdU6q3ruxEZ2yZAAAAAACAAAAAAAQZgAAAAEAACAAAABOWze7vCAhxSiq7P-B-qGuQI75QkTlc94ZpQVyFl5J7gAAAAAOgAAAAAIAACAAAABGu2Yvlo1Ifgl9rF4IkIVSrc31luJ2iQeXfG-RtfnzxHAAAACO4ZTdzoP0qHmQa2agUPhI7ZuxaSFx4gcv77g4RoUyDrTLZEW2UoVc8Qjx0i1ND2JdrMqiGASgI8lrCXWL97uR4Atvsn74ku62IQ4OlDtQDQiY9Pu5P7TQ108bXPd47EvmyKMFUjapBbUgulRkTBpoQAAAAAMeTlLhmw6StoYoUVE_cv3F2num1C1MS7focWGvftNSNeRnoaE8vwsDERDou5Ty77VmSXKOdUKkhsM1WemM5CM%26RedirectTo%3Dhttps%253A%252F%252Ffourhands-test.sandbox.operations.dynamics.com%252F&response_mode=form_post&nonce=638962306249100599.Zjc2MDkwNDEtYTQxMC00MjY4LThkN2YtMTI0MWM4ODUzZWQwYzVkZmZlMDQtMmM0OC00YjI1LTlkYzktM2M4OWY4OTEwZWYx&redirect_uri=https%3A%2F%2Ffourhands-test.sandbox.operations.dynamics.com%2F&max_age=86400&x-client-SKU=ID_NET472&x-client-ver=8.3.0.0","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"base_url","time":{"start":1760633819136,"stop":1760633819137,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"_verify_url","time":{"start":1760633819137,"stop":1760633819137,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"pytestconfig","time":{"start":1760633819137,"stop":1760633819137,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"delete_output_dir","time":{"start":1760633819137,"stop":1760633819137,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"playwright","time":{"start":1760633819138,"stop":1760633819443,"duration":305},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"playwright_browser","time":{"start":1760633819444,"stop":1760633823730,"duration":4286},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"browser_context_args","time":{"start":1760633823730,"stop":1760633823731,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"configure_browserstack","time":{"start":1760633823731,"stop":1760633823732,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"second_fixture","time":{"start":1760633823731,"stop":1760633823731,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"caplog","time":{"start":1760633823731,"stop":1760633823731,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"context","time":{"start":1760633823732,"stop":1760633823748,"duration":16},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"page","time":{"start":1760633823748,"stop":1760633823789,"duration":41},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"\n    Record D365 authentication session.\n    \n    IMPORTANT: Run this test manually with headed mode:\n        pytest tests/d365/test_d365_auth.py -k record --headed -s\n    \n    Steps:\n        1. Navigate to D365\n        2. Login manually\n        3. Wait for dashboard to load\n        4. Session is saved automatically\n    ","status":"failed","statusMessage":"Failed: Authentication timeout. Please login within 2 minutes. Error: Not on D365 dashboard yet. Current URL: https://login.microsoftonline.com/ba4f0c17-6451-450e-8adb-a7df399faaa9/oauth2/authorize?client_id=00000015-0000-0000-c000-000000000000&response_type=code%20id_token&scope=openid%20profile&state=OpenIdConnect.AuthenticationProperties%3DAQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAKa_EB3DXdU6q3ruxEZ2yZAAAAAACAAAAAAAQZgAAAAEAACAAAABOWze7vCAhxSiq7P-B-qGuQI75QkTlc94ZpQVyFl5J7gAAAAAOgAAAAAIAACAAAABGu2Yvlo1Ifgl9rF4IkIVSrc31luJ2iQeXfG-RtfnzxHAAAACO4ZTdzoP0qHmQa2agUPhI7ZuxaSFx4gcv77g4RoUyDrTLZEW2UoVc8Qjx0i1ND2JdrMqiGASgI8lrCXWL97uR4Atvsn74ku62IQ4OlDtQDQiY9Pu5P7TQ108bXPd47EvmyKMFUjapBbUgulRkTBpoQAAAAAMeTlLhmw6StoYoUVE_cv3F2num1C1MS7focWGvftNSNeRnoaE8vwsDERDou5Ty77VmSXKOdUKkhsM1WemM5CM%26RedirectTo%3Dhttps%253A%252F%252Ffourhands-test.sandbox.operations.dynamics.com%252F&response_mode=form_post&nonce=638962306249100599.Zjc2MDkwNDEtYTQxMC00MjY4LThkN2YtMTI0MWM4ODUzZWQwYzVkZmZlMDQtMmM0OC00YjI1LTlkYzktM2M4OWY4OTEwZWYx&redirect_uri=https%3A%2F%2Ffourhands-test.sandbox.operations.dynamics.com%2F&max_age=86400&x-client-SKU=ID_NET472&x-client-ver=8.3.0.0","statusTrace":"tests/d365/test_d365_auth.py:55: in test_record_d365_authentication\n    raise Exception(f\"Not on D365 dashboard yet. Current URL: {current_url}\")\nE   Exception: Not on D365 dashboard yet. Current URL: https://login.microsoftonline.com/ba4f0c17-6451-450e-8adb-a7df399faaa9/oauth2/authorize?client_id=00000015-0000-0000-c000-000000000000&response_type=code%20id_token&scope=openid%20profile&state=OpenIdConnect.AuthenticationProperties%3DAQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAKa_EB3DXdU6q3ruxEZ2yZAAAAAACAAAAAAAQZgAAAAEAACAAAABOWze7vCAhxSiq7P-B-qGuQI75QkTlc94ZpQVyFl5J7gAAAAAOgAAAAAIAACAAAABGu2Yvlo1Ifgl9rF4IkIVSrc31luJ2iQeXfG-RtfnzxHAAAACO4ZTdzoP0qHmQa2agUPhI7ZuxaSFx4gcv77g4RoUyDrTLZEW2UoVc8Qjx0i1ND2JdrMqiGASgI8lrCXWL97uR4Atvsn74ku62IQ4OlDtQDQiY9Pu5P7TQ108bXPd47EvmyKMFUjapBbUgulRkTBpoQAAAAAMeTlLhmw6StoYoUVE_cv3F2num1C1MS7focWGvftNSNeRnoaE8vwsDERDou5Ty77VmSXKOdUKkhsM1WemM5CM%26RedirectTo%3Dhttps%253A%252F%252Ffourhands-test.sandbox.operations.dynamics.com%252F&response_mode=form_post&nonce=638962306249100599.Zjc2MDkwNDEtYTQxMC00MjY4LThkN2YtMTI0MWM4ODUzZWQwYzVkZmZlMDQtMmM0OC00YjI1LTlkYzktM2M4OWY4OTEwZWYx&redirect_uri=https%3A%2F%2Ffourhands-test.sandbox.operations.dynamics.com%2F&max_age=86400&x-client-SKU=ID_NET472&x-client-ver=8.3.0.0\n\nDuring handling of the above exception, another exception occurred:\ntests/d365/test_d365_auth.py:58: in test_record_d365_authentication\n    pytest.fail(f\"Authentication timeout. Please login within 2 minutes. Error: {e}\")\nE   Failed: Authentication timeout. Please login within 2 minutes. Error: Not on D365 dashboard yet. Current URL: https://login.microsoftonline.com/ba4f0c17-6451-450e-8adb-a7df399faaa9/oauth2/authorize?client_id=00000015-0000-0000-c000-000000000000&response_type=code%20id_token&scope=openid%20profile&state=OpenIdConnect.AuthenticationProperties%3DAQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAKa_EB3DXdU6q3ruxEZ2yZAAAAAACAAAAAAAQZgAAAAEAACAAAABOWze7vCAhxSiq7P-B-qGuQI75QkTlc94ZpQVyFl5J7gAAAAAOgAAAAAIAACAAAABGu2Yvlo1Ifgl9rF4IkIVSrc31luJ2iQeXfG-RtfnzxHAAAACO4ZTdzoP0qHmQa2agUPhI7ZuxaSFx4gcv77g4RoUyDrTLZEW2UoVc8Qjx0i1ND2JdrMqiGASgI8lrCXWL97uR4Atvsn74ku62IQ4OlDtQDQiY9Pu5P7TQ108bXPd47EvmyKMFUjapBbUgulRkTBpoQAAAAAMeTlLhmw6StoYoUVE_cv3F2num1C1MS7focWGvftNSNeRnoaE8vwsDERDou5Ty77VmSXKOdUKkhsM1WemM5CM%26RedirectTo%3Dhttps%253A%252F%252Ffourhands-test.sandbox.operations.dynamics.com%252F&response_mode=form_post&nonce=638962306249100599.Zjc2MDkwNDEtYTQxMC00MjY4LThkN2YtMTI0MWM4ODUzZWQwYzVkZmZlMDQtMmM0OC00YjI1LTlkYzktM2M4OWY4OTEwZWYx&redirect_uri=https%3A%2F%2Ffourhands-test.sandbox.operations.dynamics.com%2F&max_age=86400&x-client-SKU=ID_NET472&x-client-ver=8.3.0.0","steps":[],"attachments":[{"uid":"35732192b26f009e","name":"Screenshot - test_record_d365_authentication","source":"35732192b26f009e.png","type":"image/png","size":166987},{"uid":"5bb01929bed97e5f","name":"Page HTML","source":"5bb01929bed97e5f.html","type":"text/html","size":85363},{"uid":"2af816eb1b5165cc","name":"Page URL","source":"2af816eb1b5165cc.txt","type":"text/plain","size":1038},{"uid":"94611d0515c3dbf5","name":"Test Video","source":"94611d0515c3dbf5.webm","type":"video/webm","size":6379310},{"uid":"11e65f53730be53d","name":"stdout","source":"11e65f53730be53d.txt","type":"text/plain","size":532}],"parameters":[],"stepsCount":0,"shouldDisplayMessage":true,"attachmentsCount":5,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"page::0","time":{"start":1760633945846,"stop":1760633945848,"duration":2},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"context::0","time":{"start":1760633945848,"stop":1760633945850,"duration":2},"status":"broken","statusMessage":"playwright._impl._errors.TargetClosedError: Tracing.stop: Target page, context or browser has been closed\n","statusTrace":"  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 231, in __call__\n    return self._fixture_function(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/_pytest/fixtures.py\", line 939, in _teardown_yield_fixture\n    next(it)\n  File \"/home/runner/work/QA-playwright-automation/QA-playwright-automation/conftest.py\", line 115, in context\n    context.tracing.stop(path=trace_path)\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/playwright/sync_api/_generated.py\", line 15237, in stop\n    return mapping.from_maybe_impl(self._sync(self._impl_obj.stop(path=path)))\n                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/playwright/_impl/_sync_base.py\", line 115, in _sync\n    return task.result()\n           ^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/playwright/_impl/_tracing.py\", line 68, in stop\n    await self._do_stop_chunk(path)\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/playwright/_impl/_tracing.py\", line 87, in _do_stop_chunk\n    result = await self._channel.send_return_as_dict(\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/playwright/_impl/_connection.py\", line 83, in send_return_as_dict\n    return await self._connection.wrap_api_call(\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.13/x64/lib/python3.11/site-packages/playwright/_impl/_connection.py\", line 558, in wrap_api_call\n    raise rewrite_error(error, f\"{parsed_st['apiName']}: {error}\") from None\n","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":true,"attachmentsCount":0,"hasContent":true,"attachmentStep":false},{"name":"second_fixture::0","time":{"start":1760633945851,"stop":1760633945851,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"caplog::0","time":{"start":1760633945851,"stop":1760633945851,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"playwright_browser::0","time":{"start":1760634136327,"stop":1760634136413,"duration":86},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"playwright::0","time":{"start":1760634136414,"stop":1760634136421,"duration":7},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"tag","value":"record"},{"name":"tag","value":"record"},{"name":"parentSuite","value":"tests.d365"},{"name":"suite","value":"test_d365_auth"},{"name":"host","value":"runnervmwhb2z"},{"name":"thread","value":"2689-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.d365.test_d365_auth"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Product defects","matchedStatuses":[],"flaky":false}],"tags":["record"]},"source":"44489cab4c8d9510.json","parameterValues":[]}